terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -124,21 +130,35 @@ def resolve_location_based_soil_match(soil_match: dict):
     )
 
 
-def resolve_location_matches_from_soil_id_result(soil_list_json: dict):
+def resolve_location_based_soil_matches(soil_list_json: dict):
     matches = []
     for match in soil_list_json["soilList"]:
         if match["id"]["rank_loc"] != "Not Displayed":
             matches.append(resolve_location_based_soil_match(match))
     return LocationBasedSoilMatches(matches=matches)
 
 
-def resolve_location_based_soil_matches(_parent, _info, latitude: float, longitude: float):
-    result = list_soils(lat=latitude, lon=longitude)
-
-    if isinstance(result, str):
+def resolve_list_output_failure(list_output: SoilListOutputData | str):
+    if isinstance(list_output, SoilListOutputData):
         return None
+    elif isinstance(list_output, str):
+        return SoilIdFailure(reason=SoilIdFailureReason.DATA_UNAVAILABLE)
+    else:
+        return SoilIdFailure(reason=SoilIdFailureReason.ALGORITHM_FAILURE)
+
+
+def resolve_location_based_result(_parent, _info, latitude: float, longitude: float):
+    try:
+        list_output = list_soils(lat=latitude, lon=longitude)
+
+        failure = resolve_list_output_failure(list_output)
+        if failure is not None:
+            return failure
 
-    return resolve_location_matches_from_soil_id_result(result.soil_list_json)
+        return resolve_location_based_soil_matches(list_output.soil_list_json)
+    except Exception:
+        logger.error(traceback.format_exc())
+        return SoilIdFailure(reason=SoilIdFailureReason.ALGORITHM_FAILURE)
```

Comment:
Question: I imagine there's no meaningful difference between this error case and the other case that returns `ALGORITHM_FAILURE`, just that in this one we'll get a log about it?

Reply:
hmm good question, i should log a different error for the other one, it's a catch-all because i feel confused by what the soil ID algorithm can return, but i should still log the confusion so it's easier to track down if that code path is getting triggered